File: problems/2269-find-the-k-beauty-of-a-number/solution.cpp
// Link: https://leetcode.cn/problems/find-the-k-beauty-of-a-number/
// Level: Easy
// Date: 2025-04-09

class Solution {
    public:
        int divisorSubstrings(int num, int k) {
            string s = to_string(num);
            int n = s.size();
            int res = 0;
            for (int i = 0; i < n - k + 1; i++) {
                int tmp = stoi(s.substr(i, k));
                if (tmp != 0 && num % tmp == 0) {
                    res++;
                }
            }
            return res;
        }
};

File: problems/2269-find-the-k-beauty-of-a-number/solution.py
# Link: https://leetcode.cn/problems/find-the-k-beauty-of-a-number/
# Level: Easy
# Date: 2025-04-09

class Solution:
    def divisorSubstrings(self, num: int, k: int) -> int:
        s = str(num)
        n = len(s)
        res = 0
        for i in range(n - k + 1):
            tmp = int(s[i:i+k])
            if tmp != 0 and num % tmp == 0:
                res += 1
        return res